### PR TITLE
Mfl/car camping

### PR DESCRIPTION
**Pull request by beingmattlevy**
Have the core functionality working for event, event_addons, addons. 
Models, controller, views

TicketRequests not yet complete. 
Views still need work to make more usable.

### CODE REVIEW

File: app/controllers/ticket_requests_controller.rb [hunk 1]
```diff
@@ -132,14 +138,17 @@ def create
         target: @ticket_request
       ).fire!
 
-      if (@event.tickets_require_approval || @ticket_request.free?) && @ticket_request.total_tickets > 1
+      if @event.tickets_require_approval && @ticket_request.total_tickets > 1
+        Rails.logger.debug { "tr approval: #{@ticket_request.inspect}" }
         redirect_to event_ticket_request_path(@event, @ticket_request),
                     notice: 'When you know your guest names, please return here and add them below.'
-      elsif !@ticket_request.all_guests_specified?
+      elsif !@ticket_request.all_guests_specified? && @ticket_request.total_tickets > 1
```
Comment (kigster) [inline app/controllers/ticket_requests_controller.rb:145]: Why are total_tickets must be > 1?

File: app/models/addon.rb [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+  def humanized_category
+    HUMANIZED_CATEGORIES[category]
+  end
+
+  def self.order_by_category
+    order(:category, :id)
+  end
+
+  def self.find_all_by_category(category)
```
Comment (kigster) [inline app/models/addon.rb:42]: If not used, remove

File: app/models/event.rb [hunk 3]
```diff
@@ -194,16 +175,68 @@ def ticket_requests_open?
...
-    early_arrival_price.positive? || late_departure_price.positive?
-  end
-
   def to_param
     return nil unless persisted?
 
     [id, slug].join('-') # 1-summer-campout-xii
   end
 
+  def build_event_addons_from_params(build_params)
```
Comment (kigster) [inline app/models/event.rb:184]: Perhaps use before_validation to create default values.

File: app/models/ticket_request.rb [hunk 4]
```diff
@@ -276,22 +266,30 @@ def refund
     end
   end
 
+  # calculate the total price for this ticket request
   def price
     return special_price if special_price
 
+    total = tickets_price
+    total += calculate_tr_event_addons_price
```
Comment (kigster) [inline app/models/ticket_request.rb:274]: Rename to `calculate_addon_price`

**Review by kigster — APPROVED** [review]

File: app/models/event.rb (hunk 3 above)
Comment (beingmattlevy) [reply app/models/event.rb:184]: [https://github.com/fnf-org/TicketBooth/issues/250](url) Opened issue for this to resolve later.

**Pull request merged**